### DISCUSSION TITLE
next record

### DISCUSSION THREAD

**Discussion by emersonamaro** [filamentphp/filament#13387] [Help] [answered]
Can someone help me
I have the following situation
my table with the following fields

protected $fillable = [
 'type_id',
 'responsible_id',
 'jobs_id',
 'sequence',
 'finished',
 'started',
 'priority'
 ];

When I change the Togglecolumn button I need to find the next sequence to update to true

Tables\Columns\ToggleColumn::make(name: 'started')->label('Started')
 ->onColor('success')
 ->offColor('danger')
 ->disabled(fn ($record): bool => !$record->finished === false)
 ->disabled(fn ($record): bool => !$record->started === false),

 Tables\Columns\IconColumn::make(name: 'finished')->boolean(),
 Tables\Columns\ToggleColumn::make(name: 'finished')->label('Finished')
 ->onColor('success')
 ->offColor('danger')
 ->disabled(fn ($record): bool => !$record->finished === false)
 ->beforeStateUpdated(function ($record, $state) {

 $test = $record->sequence + 1;

 return $record->update(['started' => $state?:true]);

 }),

**Comment by leandrocfe** [comment, marked as answer]
I think the next record would be

```php
->where('id', '>', $record->id)
->orderBy('id', 'asc')
->first();
```

**Reply by emersonamaro** [reply]
perfect, worked like clockwork

**Reply by emersonamaro** [reply]
What I did worked perfectly, however, I went to do a battery of tests and now I noticed that when it reaches the last record it shows an error

Error
PHP 8.2.20
10.48.10
Call to a member function update() on null


can you help me?

**Comment by micraux** [comment]
Hello,

> when it reaches the last record it shows an error

This is normal: when you are on the last record, there is no "next record"!

You just have to test your result and choose what to do when there is no "next record", like:

```php
if (is_null($your_result))
{
   // your code when NULL
}
```

**Reply by emersonamaro** [reply]
// Buscar a próxima linha de produção
                        $nextRecord = $record
                            ->where('id', '>', $record->id)
                            ->orderBy('id', 'asc')
                            ->first();

                        // Se existir próxima linha de produção, atualizá-la
                        if ($nextRecord) {
                            $nextRecord->update(['iniciado' => $state ?: true]);
                        }

                        return $state;
                    }),

Now it's perfect, thank you very much
